packages/provider/src/RelayClient.ts

```diff
@@ -201,7 +232,8 @@ export class RelayClient {
       throw new Error('no registered relayers')
     }
     const relayingErrors = new Map<string, Error>()
-    const auditPromises: Array<Promise<AuditResponse>> = []
+    this.auditPromises = []
```

Comment:
moving auditPromises into a member of RelayClient is very problematic: it means that part of the request is in a state variable.
It can create very strange bugs, if there are parallel calls to relayTransaction.
So either: 
- explicitly block such parallel calls (relayTransaction throws if called which it is already active)
- nicer: relayTransaction blocks until a previous call returns.
- better: not save data in the RelayClient object. might require creating a "RelayTransactionCall" object to hold such state (and maybe also host the callbacks
- even better: make relayTransaction itself work without callbacks, as a single call, so that it doesn't need such state variables, and it can be kept as method local member

Reply:
maybe this is not that difficult - only pass the postCallPromises array as parameter to the inner methods, to let them fill stuff into it.